### PR TITLE
Conditional build of cmdscript to allow compilation with old python versions (<3.10.x)

### PR DESCRIPTION
**Pull request by nitraiolo**
To resolve issue #1843 and allow compilation of all binaries on systems with old python version.

### CODE REVIEW

**Comment by iceman1001** [comment]
Nice one but it breaks on WSL,  so some minor tweaks?

File: client/src/cmdscript.c [hunk 1]
```diff
@@ -403,18 +403,23 @@ static int CmdScriptRun(const char *Cmd) {
         // hook Proxmark3 API
         PyImport_AppendInittab("_pm3", PyInit__pm3);
 #endif
+#if PY_MINOR_VERSION >= 10
```
Comment (jmichelp) [inline client/src/cmdscript.c:406]: Filtering only with the minor version might not be a good idea.

File: client/src/cmdscript.c [hunk 2]
```diff
@@ -424,6 +429,17 @@ static int CmdScriptRun(const char *Cmd) {
 
         // clean up
         PyConfig_Clear(&py_conf);
+#else
+        wchar_t *py_args[argc + 1];
+        py_args[0] = Py_DecodeLocale(filename, NULL);
```
Comment (jmichelp) [inline client/src/cmdscript.c:434]: Using `Py_DecodeLocale()` will allocate memory which needs to be freed at the end.

Comment (nitraiolo) [reply client/src/cmdscript.c:434]: There is the free at line 426.
The code is the one from commit #38cb327991501325aab6b24310e5428dc0369ab0

**Comment by nitraiolo** [comment]
> Nice one but it breaks on WSL, so some minor tweaks?

Some problem with preprocessor conditions? Can I see the error?

File: client/src/cmdscript.c (hunk 1 above)
Comment (nitraiolo) [reply client/src/cmdscript.c]: You have right! Added MAJOR too, thanks!

File: client/src/cmdscript.c [hunk 3]
```diff
@@ -424,6 +429,17 @@ static int CmdScriptRun(const char *Cmd) {
...
+        for (int i = 0; i < argc; i++) {
+            py_args[i + 1] = Py_DecodeLocale(argv[i], NULL);
```
Comment (jmichelp) [inline client/src/cmdscript.c:436]: Just noticed that the index might be off-by-one.
```c
for (int i = 0; i < argc + 1; i++) {
  py_args[i] = Py_DecodeLocale(argv[i], NULL);
```
This should be enough and will already handle filename (i.e. line 434 can ve removed)

Comment (nitraiolo) [reply client/src/cmdscript.c:436]: I've noticed it also that this code puts file name into args[0] and shifts all parameter by one. This was the way the code was working into commit #38cb327991501325aab6b24310e5428dc0369ab0. TBH I didn't dig more into this.

Comment (nitraiolo) [reply client/src/cmdscript.c:436]: @jmichelp tested with parity.py and you have right! That code was wrong!

[usb] pm3 --> script run parity.py 10 1234
[+] executing python /data/home/nitro/Downloads/NFC-RFID/iceman_builds/proxmark3_08/client/pyscripts/parity.py
[+] args '10 1234'
Traceback (most recent call last):
  File "parity.py", line 75, in <module>
    main()
  File "parity.py", line 61, in main
    numtype=  int(sys.argv[1], 10)
ValueError: invalid literal for int() with base 10: 'parity.py'

Fixing it as suggested works well

[usb] pm3 --> script run parity.py 10 1234
[+] executing python /data/home/nitro/Downloads/NFC-RFID/iceman_builds/proxmark3_08/client/pyscripts/parity.py
[+] args '10 1234'
numtype: 10
num: 1234 0x4D2
Even parity

[+] finished parity.py

**Comment by nitraiolo** [comment]
> Nice one but it breaks on WSL, so some minor tweaks?

The last commit should have fixed this too.

**Comment by nitraiolo** [comment]
This last commit is intended to help maintenance. This way the "default" when condition is not met (#else part) is always the new code.

**Comment by iceman1001** [comment]
Builds everywhere!

**Pull request merged**